src/ConcreteOperations/cartesian_product.jl

```diff
@@ -1,45 +1,103 @@
 export cartesian_product
 
 """
-    cartesian_product(P1::VPolytope, P2::VPolytope;
-                      [backend]=default_polyhedra_backend(P1))
+    cartesian_product(X::LazySet, Y::LazySet; [backend]=nothing, [algorithm]::String="vrep")
+
+Compute the Cartesian product of two sets.
+
+### Input
+
+- `X`         -- set
+- `Y`         -- another set
+- `backend`   -- (optional, default: `nothing`) the polyhedral computations backend
+- `algorithm` -- (optional, default: "vrep") the method used to transform the each set
+                 `X` and `Y` before taking the Cartesian product; choose between
+                 "vrep" (use the vertex representation) and "hrep" (use the constraint representation)
+
+### Output
+
+The `VPolytope` (if "vrep" was useed) or `HPolytope` (if "hrep" was used)
+obtained by the concrete Cartesian product of `X` and `Y`.
+
+### Notes
+
+For further information on the supported backends see
+[Polyhedra's documentation](https://juliapolyhedra.github.io/).
+"""
+function cartesian_product(X::LazySet, Y::LazySet; backend=nothing, algorithm::String="vrep")
+
+    if algorithm == "vrep"
+        Yv = VPolytope(vertices_list(Y))
+        if dim(X) == 1 # special case
+            Xv = convert(Interval, X)
```

Comment:
Here you assume convexity of `X`. I suggest to mark these places to ease the generalization to non-convex sets later.

Reply:
yes. but note that if the set cannot be converted exactly to an interval => the conversion fails with a method error,  it is different if i had used `overapproximate` which would return a wrong answer.

Reply:
i added a comment, feel free to suggest a different note

Reply:
I added a suggestion.